spec/app/ics-020-fungible-token-transfer/README.md

```diff
@@ -289,8 +377,20 @@ function onAcknowledgePacket(
   packet: Packet,
   acknowledgement: bytes) {
   // if the transfer failed, refund the tokens
-  if (!acknowledgement.success)
+  if !(acknowledgement.success) {
     refundTokens(packet)
+  } else {
+    // set metadata flag for any metadata we sent in the packet
+    // so we don't need to resend metadata for that token on this channel
+    // in the next sendPacket
+    FungibleTokenPacketDataV2 data = packet.data
+    for token in data.tokens {
+      if !isEmpty(token.metadata) {
+        onChainDenom = constructOnChainDenom(token.trace, token.denomination)
+        setMetadataFlag(packet.sourcePort, packet.sourceChannel, onChainDenom)
+      }
+    }
```

Comment:
Should this be also wrapped in an if statement like `if channel.version === "ics20-2"`?